apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt

```diff
@@ -1071,6 +1071,15 @@ open class ChatController(var ctrl: ChatCtrl?, val ntfManager: NtfManager, val a
         chatItemSimpleUpdate(r.chatItem)
       is CR.ChatItemDeleted -> {
         AudioPlayer.stop(r.deletedChatItem.chatItem)
+        val isLastChatItem = chatModel.getChat(r.deletedChatItem.chatInfo.id)?.chatItems?.lastOrNull()?.id == r.deletedChatItem.chatItem.id
+        if (isLastChatItem && ntfManager.hasNotificationsForChat(r.deletedChatItem.chatInfo.id)) {
+          ntfManager.cancelNotificationsForChat(r.deletedChatItem.chatInfo.id)
```

Comment:
does it cancel notifications even for undeleted items?

Reply:
and it doesn't seem to remove notification for deleted item if it wasn't the last.. correct me if I'm wrong

Reply:
> does it cancel notifications even for undeleted items?

What do you call "undeleted items"?

> and it doesn't seem to remove notification for deleted item if it wasn't the last

It's expressed by `isLastChatItem`, yes. 
So, if you received message 1 and then message 2, then the sender deleted 1, notification will still be visible for 2. There is only one message at a time that is visible.

Reply:
> There is only one message at a time that is visible.

yep, that's the part I've missed